pilot/pkg/monitoring/monitoring.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+type (
```

Comment:
Will we/should we bring this abstraction to the pkg level and use it in other components? There is no immediate benefit to going through and refactoring all of them, just wondering the intended scope of this

Reply:
Ideally, yes. I think there will need to be some tweaks as we expand this to more of the instrumented code, but I think it is close to what an `istio.io/pkg` monitoring API could look like.